.github/workflows/ci.yaml

```diff
@@ -0,0 +1,25 @@
+name: Compile, lint and test
+
+on: push
```

Comment:
What do you think to trigger the pr only when exists a [pull-request](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)?

Reply:
yeah, we can cut the executions this way!